### PR TITLE
feat(sbb-clock): component refactoring

### LINKED ISSUE #1243: story(sbb-clock): review/refactor
### Feature Description
We want to review/refactor the sbb-clock implementation.

https://lyne.sbb.ch/components/sbb-clock

[Figma Spec](https://www.figma.com/file/9r6xSfNmEfCFxl1yFYedrj/Lyne-Components?node-id=836%3A663362)

**References**

-

**Design Spec**

- Initial time is reset on tab switch/visibility switch/pause. Purpose @mcilurzo?

**Technical Spec**

- Check implementation and refactor where necessary
- Remove `paused` and `initialTime` properties and refactor related logic
  - Add possibility to add `data-now="1674125992598"` where the number is a timestamp, which sets a fixed time for the clock (it does not update)
- Ensure the sizing and position is always correct (there is a hard to reproduce bug that moves the hands into weird positions)
- Verify orientation change bug (changing orientation in the sbb-footer story on a smartphone from portrait to landscape and back)

```ts
// Example for current time
private _now(): number {
  if (this._element.dataset?.now) {
    const dataNow = +this._element.dataset?.now;
    return isNaN(dataNow) ? Date.now() : dataNow;
  }
  return Date.now();
}
```

### Definition of Done
 - [ ]  Component is implemented
 - [ ]  Documentation is provided
 - [ ]  Tests are implemented
 - [ ]  E2E Tests are implemented
 - [ ]  Storybook stories are implemented
 - [ ]  Navigation via keyboard is tested
 - [ ]  Screen reader output is tested
 - [ ]  High-contrast is tested
 - [ ]  Remaining accessibility is tested
 - [ ]  UX approved

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #1243

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- component refactoring

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [x] Edge Desktop
- [x] Safari Desktop
- [x] Chrome Mobile
- [x] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): component review

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

- properties `paused` and `initialTime` have been removed;
- for testing purpose, the `data-now` attribute (time in millisecond) can be used to stop the clock at a specific time.

### CODE REVIEW

File: src/components/sbb-clock/sbb-clock.stories.js [hunk 1]
```diff
@@ -2,77 +2,35 @@ import { h } from 'jsx-dom';
...
 export const Paused = Template.bind({});
-
-Paused.argTypes = defaultArgTypes;
-Paused.args = {
-  'initial-time': times.options[1],
-  paused: true,
-};
-
+Paused.argTypes = { 'data-now': dataNow };
+Paused.args = { 'data-now': 1677273030000 };
```
Comment (jeripeierSBB) [inline src/components/sbb-clock/sbb-clock.stories.js:22]: I'd prefer the approach like you did on line 33

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
you need to update footer and home stories with the data-now attribute

File: src/components/sbb-clock/sbb-clock.stories.js [hunk 2]
```diff
@@ -2,77 +2,35 @@ import { h } from 'jsx-dom';
 import isChromatic from 'chromatic/isChromatic';
 import readme from './readme.md';
 
-const Template = (args) => <sbb-clock {...args} />;
-
-const times = {
+const dataNow = {
   control: {
-    type: 'select',
+    type: 'number',
```
Comment (jeripeierSBB) [inline src/components/sbb-clock/sbb-clock.stories.js:7]: I think you can use the same approach as in other components where you can chose a date (date control or similar)

File: src/components/sbb-clock/sbb-clock.tsx [hunk 3]
```diff
@@ -89,132 +103,127 @@ export class SbbClock implements ComponentInterface {
...
     this._clockHandSeconds?.classList.remove('sbb-clock__hand-seconds--initial-minute');
     this._clockHandMinutes?.classList.remove('sbb-clock__hand-minutes--no-transition');
-    this._element.style.removeProperty('--clock-seconds-animation-start-angle');
-    this._element.style.removeProperty('--clock-seconds-animation-duration');
+    this._element.style.removeProperty('--sbb-clock-seconds-animation-start-angle');
+    this._element.style.removeProperty('--sbb-clock-seconds-animation-duration');
   }
 
+  /** Given the current date, calculates the hh/mm/ss values and the hh/mm/ss left to the next midnight. */
   private _getCurrentTime(): void {
```
Comment (kyubisation) [inline src/components/sbb-clock/sbb-clock.tsx:118]: Nitpick; This does not actually get the current time, but assign the current time. Should this maybe be renamed to `_assignCurrentTime`?

File: src/components/sbb-clock/sbb-clock.tsx [hunk 4]
```diff
@@ -89,132 +103,127 @@ export class SbbClock implements ComponentInterface {
...
+  /** Move the minutes hand to the next value. */
   private _moveMinutesHand(): void {
     this._clockHandSeconds?.removeEventListener('animationend', moveMinutesHand);
 
     this._removeSecondsAnimationStyles();
 
+    this._addMinutesAndSetHands();
+
+    handMovement = setInterval(
+      this._addMinutesAndSetHands.bind(this),
```
Comment (kyubisation) [inline src/components/sbb-clock/sbb-clock.tsx:211]: Nitpick; I generally prefer `() => this.method()` to `this.method.bind(this)` as it is easier to find in my mind.
What do you think?

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Excellent work!

File: src/components/sbb-clock/sbb-clock.stories.js [hunk 5]
```diff
@@ -2,77 +2,35 @@ import { h } from 'jsx-dom';
...
  */
 if (isChromatic()) {
   Default.args = {
-    'initial-time': times.options[1],
-    paused: true,
-  };
-
-  InitialTime.args = {
-    paused: true,
+    'data-now': new Date('2023-01-24T10:10:30+01:00').valueOf(),
```
Comment (jeripeierSBB) [inline src/components/sbb-clock/sbb-clock.stories.js:33]: I see that the clock is displayed with 09:10:30. I guess it's coming from the +01:00 timezone.  But the question is, in which timezone should the clock be shown? As it's only for testing, this is not a big question, but I'm insecure wether the current approach is the best?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Sadly, the iOS bug still exists

File: src/components/sbb-clock/sbb-clock.stories.js (hunk 5 above)
Comment (jeripeierSBB) [reply src/components/sbb-clock/sbb-clock.stories.js:33]: It's only chromatic. All is ok

**Comment by DavideMininni-Fincons** [comment]
> Sadly, the iOS bug still exists

@jeripeierSBB is there a way to replicate this? Tried again this morning in Safari, both Mac and iPhone, and it seems to work, I've tried app switching, rotating, waiting for minute change, and so on.. It only looks a bit 'slower' on iPhone, but maybe it's just a feeling

**Comment by jeripeierSBB** [comment]
> > Sadly, the iOS bug still exists
> 
> @jeripeierSBB is there a way to replicate this? Tried again this morning in Safari, both Mac and iPhone, and it seems to work, I've tried app switching, rotating, waiting for minute change, and so on.. It only looks a bit 'slower' on iPhone, but maybe it's just a feeling

I'll have a look at it. I took the home story in portrait mode, changed to landscape mode and back to portrait mode and then clock hands were misplaced all the time. (iPhone 13, newest iOS)

**Review by jeripeierSBB — APPROVED** [review]
LGTM, thanks!

**Pull request merged**